include/langsvr/traits.h

```diff
@@ -31,6 +31,12 @@
 #include <string>
 #include <tuple>
 
+/// Forward declaration
+namespace langsvr {
+template <typename SUCCESS_TYPE, typename FAILURE_TYPE>
+struct Result;
+}
```

Comment:
Doesn't clang format add `// namespace langsvr` on the closing brace ususally?

Reply:
I think it omits it if the open and close are close.